iron-ajax.html

```diff
@@ -452,6 +457,20 @@
         this._discardRequest.bind(this, request)
       );
 
+      var event = new CustomEvent('iron-ajax-presend', {
+        detail: {
+          request: request,
+          options: requestOptions
+        },
+        bubbles: this.bubbles,
+        cancelable: true,
+        composed: true
+      });
+      var cancelled = !this.dispatchEvent(event);
+      if (cancelled) {
+        return request;
```

Comment:
How do we want this handled in the cancelled scenario?

just returning the `request` will result in the promise chain not triggering. I'm unsure if we would want the promise chain to resolve or reject and how to go about doing that. Also would we want to call the discardRequest method?